jvm-packages/xgboost4j/src/main/scala/ml/dmlc/xgboost4j/scala/DeviceQuantileDMatrix.scala

```diff
@@ -0,0 +1,107 @@
+/*
+ Copyright (c) 2014 by Contributors
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+package ml.dmlc.xgboost4j.scala
+
+import _root_.scala.collection.JavaConverters._
+
+import ml.dmlc.xgboost4j.java.{Column, ColumnBatch, XGBoostError, DeviceQuantileDMatrix => JDeviceQuantileDMatrix}
+
+class DeviceQuantileDMatrix private[scala](
+  private[scala] override val jDMatrix: JDeviceQuantileDMatrix) extends DMatrix(jDMatrix) {
+
+  /**
+   * Create DeviceQuantileDMatrix from iterator based on the cuda array interface
+   *
+   * @param iter    the XGBoost ColumnBatch batch to provide the corresponding cuda array interface
+   * @param missing the missing value
+   * @param maxBin  the max bin
+   * @param nthread the parallelism
+   * @throws XGBoostError
+   */
+  def this(iter: Iterator[ColumnBatch], missing: Float, maxBin: Int, nthread: Int) {
+    this(new JDeviceQuantileDMatrix(iter.asJava, missing, maxBin, nthread))
+  }
+
+  /**
+   * set label of dmatrix
+   *
+   * @param labels labels
+   */
+  @throws(classOf[XGBoostError])
+  override def setLabel(labels: Array[Float]): Unit =
```

Comment:
Not a requirement.  As I mentioned before, would be great to eliminate most of the exception logic in a functional programming language.